flex_model/distributed/mappings.py

```diff
@@ -237,30 +238,301 @@ def scatter_data_parallel(tensor: Tensor, dim: int = 0) -> Tensor:
     return output_tensor
 
 
-def gather_pipeline_parallel(objects: Any) -> Any:
-    """Gather an object from non-zero ranks to rank0 in the pipeline group.
+def _group_by_dtype(
```

Comment:
I'm thinking that we drop docstrings for private functions, wdyt?

Reply:
I'd noticed your private functions had docstrings in other files so I left this comment. If most of your private functions in other files have docstrings, then just include them imo. Otherwise, you can leave them out. Just be sure to be consistent

Reply:
Docstrings are only mandatory on public APIs in my view. Typing I think we should aim to always populate. What do people think?